config.go

```diff
@@ -42,11 +45,40 @@ func WithServerHandshaker(handshaker Handshaker) ServerOpt {
 
 // WithUnaryServerInterceptor sets the provided interceptor on the server
 func WithUnaryServerInterceptor(i UnaryServerInterceptor) ServerOpt {
+	return func(c *serverConfig) error {
+		if c.interceptor == nil {
+			c.interceptor = i
+		} else {
+			WithChainUnaryServerInterceptor(i)(c)
+		}
+		return nil
+	}
+}
+
+// WithChainUnaryServerInterceptor sets the provided chain of server interceptors
+func WithChainUnaryServerInterceptor(interceptors ...UnaryServerInterceptor) ServerOpt {
 	return func(c *serverConfig) error {
 		if c.interceptor != nil {
-			return errors.New("only one interceptor allowed per server")
+			interceptors = append([]UnaryServerInterceptor{c.interceptor}, interceptors...)
+		}
+		c.interceptor = func(
+			ctx context.Context,
+			unmarshal Unmarshaler,
+			info *UnaryServerInfo,
+			method Method) (interface{}, error) {
+			return interceptors[0](ctx, unmarshal, info,
```

Comment:
what if the `interceptors` is empty?

Reply:
I added a length check to protect against that.